File: examples/python/core/child-contexts/send-notifications.py
@durable_with_child_context
def send_notifications(ctx: DurableContext, user_id: str, message: str) -> dict:
    """Send notifications through multiple channels."""
    email_sent = ctx.step(
        lambda _: send_email(user_id, message),
        name="send_email"
    )
    
    sms_sent = ctx.step(
        lambda _: send_sms(user_id, message),
        name="send_sms"
    )
    
    push_sent = ctx.step(
        lambda _: send_push_notification(user_id, message),
        name="send_push"
    )
    
    return {
        "email": email_sent,
        "sms": sms_sent,
        "push": push_sent,
    }

@durable_execution
def order_confirmation_handler(event: dict, context: DurableContext) -> dict:
    """Send order confirmation notifications."""
    notifications = context.run_in_child_context(
        send_notifications(
            event["user_id"],
            f"Order {event['order_id']} confirmed"
        ),
        name="order_notifications"
    )
    
    return {"notifications_sent": notifications}

@durable_execution
def shipment_handler(event: dict, context: DurableContext) -> dict:
    """Send shipment notifications."""
    notifications = context.run_in_child_context(
        send_notifications(
            event["user_id"],
            f"Order {event['order_id']} shipped"
        ),
        name="shipment_notifications"
    )
    
    return {"notifications_sent": notifications}

File: examples/java/core/child-contexts/send-notifications.java
public class OrderConfirmationHandler extends DurableHandler<Map<String, Object>, Map> {

    @Override
    public Map handleRequest(Map<String, Object> event, DurableContext ctx) {
        String userId = (String) event.get("user_id");
        String message = "Order " + event.get("order_id") + " confirmed";

        var notifications = ctx.runInChildContext("order_notifications", Map.class, childCtx -> {
            var emailSent = childCtx.step("send_email", Boolean.class,
                stepCtx -> sendEmail(userId, message));
            var smsSent = childCtx.step("send_sms", Boolean.class,
                stepCtx -> sendSms(userId, message));
            var pushSent = childCtx.step("send_push", Boolean.class,
                stepCtx -> sendPushNotification(userId, message));

            return Map.of("email", emailSent, "sms", smsSent, "push", pushSent);
        });

        return Map.of("notifications_sent", notifications);
    }
}

public class ShipmentHandler extends DurableHandler<Map<String, Object>, Map> {

    @Override
    public Map handleRequest(Map<String, Object> event, DurableContext ctx) {
        String userId = (String) event.get("user_id");
        String message = "Order " + event.get("order_id") + " shipped";

        var notifications = ctx.runInChildContext("shipment_notifications", Map.class, childCtx -> {
            var emailSent = childCtx.step("send_email", Boolean.class,
                stepCtx -> sendEmail(userId, message));
            var smsSent = childCtx.step("send_sms", Boolean.class,
                stepCtx -> sendSms(userId, message));
            var pushSent = childCtx.step("send_push", Boolean.class,
                stepCtx -> sendPushNotification(userId, message));

            return Map.of("email", emailSent, "sms", smsSent, "push", pushSent);
        });

        return Map.of("notifications_sent", notifications);
    }
}
